dojo/tools/gitlab_dast/parser.py

```diff
@@ -0,0 +1,168 @@
+import hyperlink
+from dojo.models import Finding, Endpoint
+
+
+class GitlabDastParser(object):
+    """
+    Import GitLab DAST Report in JSON format
+    """
+
+    def get_scan_types(self):
+        return ["GitLab DAST Report"]
+
+    def get_label_for_scan_types(self, scan_type):
+        return "GitLab DAST Report"
+
+    def get_description_for_scan_types(self, scan_type):
+        return "GitLab DAST Report in JSON format (option --json)."
+
+    # turning a json file to string
+    def parse_json(self, file):
+        data = file.read()
+        try:
+            tree = json.loads(str(data, 'utf-8'))
+        except:
+            tree = json.loads(data)
+
+        return tree
+
+    def get_items(self, tree, test):
+        items = {}
+
+        # iterating through each vulnerability
+        for node in tree['vulnerabilities']:
+            item = get_item(node, test)
+            if item:
+                items[item.unique_id_from_tool] = item
+
+        return list(items.values())
+
+    def get_findings(self, file, test):
+        if file is None:
+            return None
+
+        tree = self.parse_json(file)
+        if tree:
+            return self.get_items(tree, test)
+
+    def convert_severity(self, num_severity):
+        """Convert severity value"""
+        if num_severity >= -10:
+            return "Low"
+        elif -11 >= num_severity > -26:
+            return "Medium"
+        elif num_severity <= -26:
+            return "High"
+        else:
+            return "Info"
+
+
+# iterating through properties of each vulnerability
+def get_item(vuln, test):
+
+    if vuln["category"] != "dast":
+        return None
+
+    # scanner_confidence
+    scanner_confidence = get_confidence_numeric(vuln["confidence"])
+
+    # id
+    if "id" in vuln:
+        unique_id_from_tool = vuln["id"]
+    else:  # deprecated
+        unique_id_from_tool = vuln["cve"]
+
+    # title
+    if "name" in vuln:
+        title = vuln["name"]
+    # fallback to using id as a title
+    else:
+        title = unique_id_from_tool
+
+    # description
+    description = f"Scanner: {vuln['scanner']['name']}\n"
+    if "message" in vuln:
+        description += f"{vuln['message']}\n"
+    elif "description" in vuln:
+        description += f"{vuln['description']}\n"
+
+    # date
+    if "discovered_at" in vuln:
+        date = datetime.strptime(vuln["discovered_at"], "%Y-%m-%dT%H:%M:%S.%f")
+    else:
+        date = None
+
+    # endpoint
+    location = vuln["location"]
+    if "hostname" in location and "path" in location:
+        url_str = f"{location['hostname']}{location['path']}"
+        url = hyperlink.parse(url_str)
+        endpoint = Endpoint.from_uri(url)
+        print("Using url as an arg:")
+        print("Endpoint: " + str(endpoint))
+
+        print("Using str as an arg:")
+        endpoint2 = Endpoint.from_uri(url_str)
+        print("Endpoint: " + str(endpoint2))
+    else:
+        endpoint = None
```

Comment:
this one will set the `unsaved_endpoints` of the finding to `[None]` which is invalid.